gapic/src/main/com/google/gapid/perfetto/views/TraceMetadataDialog.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2020 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gapid.perfetto.views;
+
+
+import static com.google.gapid.widgets.Widgets.createTableViewer;
+import static com.google.gapid.widgets.Widgets.packColumns;
+
+import com.google.gapid.models.Analytics.View;
+import com.google.gapid.models.Models;
+import com.google.gapid.perfetto.QueryViewer.ResultContentProvider;
+import com.google.gapid.perfetto.QueryViewer.Row;
+import com.google.gapid.proto.perfetto.Perfetto;
+import com.google.gapid.proto.service.Service.ClientAction;
+import com.google.gapid.rpc.Rpc;
+import com.google.gapid.rpc.Rpc.Result;
+import com.google.gapid.rpc.RpcException;
+import com.google.gapid.rpc.UiCallback;
+import com.google.gapid.util.Messages;
+import com.google.gapid.widgets.DialogBase;
+import com.google.gapid.widgets.Theme;
+import com.google.gapid.widgets.Widgets;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.viewers.LabelProvider;
+import org.eclipse.jface.viewers.TableViewer;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.swt.widgets.TableColumn;
+
+public class TraceMetadataDialog {
+
+  private static final Logger LOG = Logger.getLogger(TraceMetadataDialog.class.getName());
+  private static final String QUERY_STR =
+      "SELECT name AS 'Name', CAST(COALESCE(int_value, str_value, 'NULL') as TEXT) as 'Value' FROM"
+          + " metadata;";
+
+  public static void showMetadata(Shell shell, Models models, Theme theme) {
+    models.analytics.postInteraction(View.Help, ClientAction.Show);
```

Comment:
The analytics should record a new action here, maybe a new item View.QueryMetadata.